src/encoded/tests/test_structured_data.py

```diff
@@ -0,0 +1,592 @@
+from contextlib import contextmanager
+import inspect
+import json
+import os
+import pdb
+import re
+from typing import Callable, List, Optional, Union
+from unittest import mock
+from dcicutils.bundle_utils import RefHint
+from dcicutils.misc_utils import to_camel_case
+from dcicutils.validation_utils import SchemaManager  # noqa
+from dcicutils.zip_utils import temporary_file
+from encoded.ingestion.structured_data import get_type_name, Portal, Schema # noqa
+from encoded.ingestion.ingestion_processors import parse_structured_data
+
+THIS_TEST_MODULE_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+TEST_FILES_DIR = f"{THIS_TEST_MODULE_DIRECTORY}/data/test-files"
+SAME_AS_EXPECTED_REFS = {}
+SAME_AS_NOREFS = {}
+
+
+def test_parse_structured_data_0():
+    _test_parse_structured_data(sheet_utils_also = True, noschemas = True,
+        as_file_name = "some_test.csv",
+        rows = [
+            r"uuid,status,principals.view,principals.edit,extensions#,data",
+            r"some-uuid-a,public,pav-a,pae-a,alfa|bravo|charlie,123.4",
+            r"some-uuid-b,public,pav-b,pae-b,delta|echo|foxtrot|golf,xyzzy"
+        ],
+        expected = {
+            "SomeTest": [
+                {
+                    "uuid": "some-uuid-a",
+                    "status": "public",
+                    "principals": { "view": "pav-a", "edit": "pae-a"
+                },
+                    "extensions": [ "alfa", "bravo", "charlie" ],
+                    "data": "123.4"
+                },
+                {
+                    "uuid": "some-uuid-b",
+                    "status": "public",
+                    "principals": { "view": "pav-b", "edit": "pae-b" },
+                    "extensions": [ "delta", "echo", "foxtrot", "golf" ],
+                    "data": "xyzzy"
+                }
+            ]
+        }
+    )
+
+def test_parse_structured_data_1():
+    _test_parse_structured_data(sheet_utils_also = False,
+        as_file_name = "some_test.csv",
+        rows = [
+            r"uuid,status,principals.view,principals.edit,extensions#,num,i,arr",
+            r"some-uuid-a,public,pav-a,pae-a,alfa|bravo|charlie,123.4,617,hotel",
+            r"some-uuid-b,public,pav-b,pae-b,delta|echo|foxtrot|golf,987,781,indigo\|juliet|kilo"
```

Comment:
This is mostly a stylistic comment, but seeing how many of these there are I do think you should consider using `@pytest.mark.parametrize` to do most of these no? It will do this sort of numbering thing for you. Because of the data size it might be less clean in practice but I think overall may be easier to see what's going on in the big picture.

Reply:
Ah, I didn't realize that benefit, of numbering output - I will check it out, thanks.